runtime/vm/resolvesupport.cpp

```diff
@@ -1847,6 +1847,113 @@ resolveMethodHandleRef(J9VMThread *vmThread, J9ConstantPool *ramCP, UDATA cpInde
 
 }
 
+j9object_t
+resolveConstantDynamic(J9VMThread *vmThread, J9ConstantPool *ramCP, UDATA cpIndex, UDATA resolveFlags)
+{
+	J9RAMConstantDynamicRef *ramCPEntry = (J9RAMConstantDynamicRef*)ramCP + cpIndex;
+	j9object_t value = NULL;
+	J9JavaVM *vm = vmThread->javaVM;
+	bool resolved = false;
+
+	/* Fast path resolution for previously resolved Entries */
+	if (NULL != ramCPEntry->value) {
+		value = ramCPEntry->value;
+		resolved = true;
+	} else {
+		/* Slow path resolve */
+		omrthread_monitor_enter(vm->constantDynamicMutex);
+retry:
+		/* Check if already resolved */
+		if (NULL != ramCPEntry->value) {
+			value = ramCPEntry->value;
+			resolved = true;
+		} else if (NULL != ramCPEntry->exception) {
+			J9Class *throwable = J9VMJAVALANGTHROWABLE_OR_NULL(vm);
+			if (isSameOrSuperClassOf(throwable, J9VM_J9CLASS_FROM_HEAPCLASS(vmThread, ramCPEntry->exception))) {
```

Comment:
J9VM_J9CLASS_... works on class objects - if exception is NULL, this will crash, and if it's a thread object, the results are undefined at best.  At the very least, the threadObject check needs to be first.